plugins/org.eclipse.n4js/src/org/eclipse/n4js/binaries/Binary.java

```diff
@@ -107,37 +137,67 @@ default String getDescription() {
 	 *            a map of key-values representing the environment.
 	 * @return the argument.
 	 */
-	Map<String, String> updateEnvironment(Map<String, String> environment);
+	public Map<String, String> updateEnvironment(Map<String, String> environment) {
+		final String additionalPath = getBinaryDirectory();
+		final String actualPathPropertyName = findActualPropertyNameOrDefault(environment, PATH);
+		final String newValue = environment.containsKey(actualPathPropertyName)
+				? environment.get(actualPathPropertyName) + File.pathSeparator + additionalPath
+				: additionalPath;
+
+		environment.put(actualPathPropertyName, newValue);
+		return environment;
+	}
+
+	/** Deals with case-insensitivity on environment variables on windows platform */
+	protected String findActualPropertyNameOrDefault(Map<String, String> environment, String defaultName) {
+		if (environment.containsKey(defaultName)) {
+			return defaultName;
+		}
+		if (OSInfo.isWindows()) {
+			if (System.getenv(PATH) == null) {
+				return defaultName;
```

Comment:
I don't understand the rationale for this case. Why does the return value of such a general-purpose utility method depend on the specific environment variable "PATH"?